src/dict.c

```diff
@@ -951,6 +952,15 @@ unsigned long dictScan(dict *d,
 
 /* ------------------------- private functions ------------------------------ */
 
+/* Check if we can expand dict. We call this when add entries into dict and
+ * generally need to expand dict. */
+static int dictCanExpandCheck(unsigned long new_size) {
```

Comment:
```
#define ALWAYS_EXPAND_MAX_SIZE;
if (new_size <= ALWAYS_EXPAND_MAX_SIZE) return 1;
```
like this?

Reply:
i'm not sure..  maybe that's a valid approach, hope to get more feedback from others before we decide.